File: 11-Bridge/src/go/radio.go
package main

import "fmt"

type Radio struct {
	enabled bool
	volume  int
	channel int
}

func NewRadio() *Radio {
	return &Radio{enabled: false, volume: 20, channel: 88}
}

func (r *Radio) IsEnabled() bool { return r.enabled }

func (r *Radio) Enable() {
	r.enabled = true
	fmt.Println("  Radio is ON")
}

func (r *Radio) Disable() {
	r.enabled = false
	fmt.Println("  Radio is OFF")
}

func (r *Radio) GetVolume() int { return r.volume }

func (r *Radio) SetVolume(volume int) {
	if volume < 0 {
		volume = 0
	} else if volume > 100 {
		volume = 100
	}
	r.volume = volume
	fmt.Printf("  Radio volume: %d\n", r.volume)
}

func (r *Radio) GetChannel() int { return r.channel }

func (r *Radio) SetChannel(channel int) {
	r.channel = channel
	fmt.Printf("  Radio station: %d FM\n", r.channel)
}

func (r *Radio) GetName() string { return "Radio" }

File: 11-Bridge/src/rust/radio.rs
use crate::device::Device;

pub struct Radio {
    enabled: bool,
    volume: i32,
    channel: i32,
}

impl Radio {
    pub fn new() -> Self {
        Radio { enabled: false, volume: 20, channel: 88 }
    }
}

impl Device for Radio {
    fn is_enabled(&self) -> bool { self.enabled }

    fn enable(&mut self) {
        self.enabled = true;
        println!("  Radio is ON");
    }

    fn disable(&mut self) {
        self.enabled = false;
        println!("  Radio is OFF");
    }

    fn get_volume(&self) -> i32 { self.volume }

    fn set_volume(&mut self, volume: i32) {
        self.volume = volume.clamp(0, 100);
        println!("  Radio volume: {}", self.volume);
    }

    fn get_channel(&self) -> i32 { self.channel }

    fn set_channel(&mut self, channel: i32) {
        self.channel = channel;
        println!("  Radio station: {} FM", self.channel);
    }

    fn get_name(&self) -> &str { "Radio" }
}
